packages/search-input/src/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,408 @@
+            key: `result-${index}`,
+            ref: child.props.ref ?? resultRefs?.(`${index}`),
+            focused: index === highlightIndex,
+            onClick: onElementClick,
+          });
+        } else if (isComponentType(child, 'SearchResultGroup')) {
+          const nestedChildren = React.Children.map(
+            child.props.children,
+            processChild,
+          );
+
+          if (nestedChildren && nestedChildren.length > 0) {
+            return React.cloneElement(child, {
+              ...child.props,
+              children: nestedChildren,
+            });
+          }
+        }
+      };
+
+      const updatedChildren = React.Children.map(children, processChild);
+
+      return {
+        resultsCount,
+        updatedChildren,
+      };
+    }, [children, highlightIndex, inputRef, resultRefs, setInputValue]);
+
+    const { updatedChildren, resultsCount } = useMemo(
+      () => processChildren(),
+      [processChildren],
+    );
+
+    type Direction = 'next' | 'prev' | 'first' | 'last';
+    const updateHighlight = (direction: Direction) => {
+      switch (direction) {
+        case 'first': {
+          setHighlightIndex(0);
+          break;
+        }
+
+        case 'last': {
+          setHighlightIndex(resultsCount);
+          break;
+        }
+
+        case 'next': {
+          const nextIndex =
+            !isUndefined(highlightIndex) && highlightIndex + 1 < resultsCount
+              ? highlightIndex + 1
+              : 0;
+          setHighlightIndex(nextIndex);
+          break;
+        }
+
+        case 'prev': {
+          const nextIndex =
+            !isUndefined(highlightIndex) && highlightIndex - 1 >= 0
+              ? highlightIndex - 1
+              : resultsCount - 1;
+          setHighlightIndex(nextIndex);
+        }
+      }
+    };
+
+    const isElementInComponent = (el: Element | null) => {
+      return formRef.current?.contains(el) || menuRef.current?.contains(el);
+    };
+
+    /** Event Handlers */
+
+    const handleOpenMenuAction: EventHandler<SyntheticEvent<any>> = e => {
+      if (disabled) {
+        e.preventDefault();
+        e.stopPropagation();
+      } else {
+        openMenu();
+      }
+    };
+
+    const handleSearchBoxMousedown = (e: React.MouseEvent) => {
+      if (disabled) {
+        // Prevent container from gaining focus by default
+        e.preventDefault();
+      }
+    };
+
+    const handleSearchBoxClick: MouseEventHandler = handleOpenMenuAction;
+
+    // Fired whenever the wrapper gains focus,
+    // and any time the focus within changes
+    const handleSearchBoxFocus: FocusEventHandler = e => {
+      const target =
+        e.target !== clearButtonRef.current
+          ? inputRef.current ?? (e.target as HTMLElement)
+          : clearButtonRef.current;
+      target.focus();
+      trackFocusedElement(target);
+      handleOpenMenuAction(e);
+    };
+
+    const handleClearButton: MouseEventHandler<HTMLButtonElement> = e => {
```

Comment:
I'm not seeing a clear button rendered in storybook ATM. When you click an item, the menu closes but the search bar isn't updated

Reply:
Selecting an item is only partially working at the moment

Reply:
Fixed the underlying issue here. The clear button will only appear if there's text in the input